### PR TITLE
New Chaplain Weapon: Hammer of Pride!

### PR DESCRIPTION
**Pull request by ghost**
It's a hammer that, when you attack others, will transfer your reagents to them. Only really useful if you poison yourself then attack them, but I'm sure people will find fun uses like moving over healing reagents or something like that.

If you don't merge this by tuesday i'll give you some unique sprites myself, if not i'll just PR the new sprite when I have time

a lot of credit to sawcalls or whomever for coding this, it's from his bishop of sin.

### CODE REVIEW

**Comment by Armhulen** [comment]
I'll come back to holy crunch when the world is ready for it!

File: code/game/objects/items/holy_weapons.dm [hunk 1]
```diff
@@ -294,6 +294,27 @@
...
+	hitsound = 'sound/weapons/blade1.ogg'
+	
+/obj/item/nullrod/pride_hammer/afterattack(atom/A as mob|obj|turf|area, mob/user, proximity)
+	if(!proximity)
+		return
+	if(istype(A,/mob/living/carbon/human))
+		var/mob/living/carbon/human/H = A
+		if(H)
+			user.reagents.trans_to(H, user.reagents.total_volume, 1, 1, 0)
+			user << "Your pride reflects on [H]."
```
Comment (MoreRobustThanYou) [inline code/game/objects/items/holy_weapons.dm]: to_chat!

**Comment by Armhulen** [comment]
fixed

**Comment by Ratvar** [comment]
Slimepeople transfering toxins to others? Will it work through thick suits?

**Comment by FantasticFwoosh** [comment]
So melee syringes with piercing protection unless you wear a riot shield all the time (???)

Bad, downthumbed, i feel sorry for downstreams for being subjected to this validhunting chaplain weapon bloat.

**Comment by Armhulen** [comment]
To inject them with poison you have to inject yourself with poison and if you have any healing reagents you give it to them

Fwoosh posting for fucks sake 👀

**Comment by XDTM** [comment]
he's right to be honest, you can down a 50u pill of poison and only suffer one tick while injecting the rest

**Comment by FantasticFwoosh** [comment]
Just be a plasmaman with Toxin loving, or be very very drunk.

**Comment by Armhulen** [comment]
Sounds pretty smart

I'd be okay with lowering the damage to 16 but if you're planning beforehand a fight and you down the pill then get off a hit it's a pretty good payoff. How to counter this? Just run away when the chaplain downs the pill, he has to deal with the 50u of poison pretty fast sooo

**Comment by FantasticFwoosh** [comment]
Dental implants BOOM

**Comment by Armhulen** [comment]
And beyond that, I'll give this a distinct sprite when I'm able to so you'll know if it's a hammer chappy

**Comment by FantasticFwoosh** [comment]
There's already a hammer chap

File: code/game/objects/items/holy_weapons.dm [hunk 2]
```diff
@@ -294,6 +294,27 @@
...
+	throwforce = 15
+	w_class = 4
+	slot_flags = SLOT_BACK
+	attack_verb = list("attacked", "smashed", "crushed", "splattered", "cracked")
+	hitsound = 'sound/weapons/blade1.ogg'
+	
+/obj/item/nullrod/pride_hammer/afterattack(atom/A as mob|obj|turf|area, mob/user, proximity)
+	if(!proximity)
+		return
+	if(istype(A,/mob/living/carbon/human))
```
Comment (ShizCalev) [inline code/game/objects/items/holy_weapons.dm]: if(ishuman(A))

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/game/objects/items/holy_weapons.dm [hunk 3]
```diff
@@ -294,6 +294,27 @@
...
+		var/mob/living/carbon/human/H = A
+		if(H)
```
Comment (ShizCalev) [inline code/game/objects/items/holy_weapons.dm]: You're already checking if they're human. Can drop this.

**Comment by Armhulen** [comment]
"there's already a sword chap"

and besides, if you go through surgery to get a deathpill added to your mouth to use before they're fighting you they deserve the extra hit.

I'll lower the damage to 16 but no further

**Comment by FantasticFwoosh** [comment]
Just fit a entire mouth of deathpills and murder everyone you meet with a mixture of neurotoxin and beepsky smash (or whatever else) with gratidous beating afterwards or a e-bow

- You KNOW this is going to be abused super hard versus wizards and other such antags, not least the fact you can make pills out of Plant X reagents too. 

Its not even nessecary to kill as long as the effect is not immediate before being passed over, since you can pop capascin/teslium right into someone then shoot them with a double tap.

**Comment by Armhulen** [comment]
Beepsky Smash would kill you.

**Comment by FantasticFwoosh** [comment]
Doesn't matter if you pop it fast enough then transfer to your target who'd die with you

**Comment by Armhulen** [comment]
Err stun you, before you could inject it

**Comment by Armhulen** [comment]
You'd have to pop it on the same tick you attack that sounds like very robust skills that we reward with good damage

**Comment by FantasticFwoosh** [comment]
See prior point about non-immediate acting or metabolisation chems then.

- Amatoxin would be the real kicker in that case, 50u of amatoxin, inject into someone then run away and get chem treatment, wait rinse and repeat.

**Comment by Armhulen** [comment]
I guess what I'm trying to say is that people don't stand still and in reality this will probably never be an issue

**Comment by FantasticFwoosh** [comment]
People stand still at the time, just do it while they are looking at computers/screens or talking to other people, no good person trying to off someone chases their target.

**Comment by Armhulen** [comment]
If you trust in roboticists to add deathchems that you either bought or the chemists gave you for some reason then they add it and you then get off your super deadly poison on the same tick you smack then yeah sure it's kinda strong I guess

**Comment by FantasticFwoosh** [comment]
It doesn't have to be same tick is what im saying, we have so many varied & deadly chemicals, ill quickly dive into the code and find a few. You're clinging onto the thought that the user will most definitely die but ingestion rates (and cheating by removing your own liver or never having one) differ between chemicals.

Polonium for one.

**Comment by Armhulen** [comment]
I guess I could make it a syndicate item

**Comment by jughu** [comment]
Make it random chance

**Comment by Armhulen** [comment]
Also fwoosh if you're mad at this just wait until you see my next chap item

**Comment by Armhulen** [comment]
Oooh random chance that's good what %

**Comment by jughu** [comment]
30% sounds like a reasonable chance

**Comment by FantasticFwoosh** [comment]
Then its just a gimmick like giving a second bible, what objective use would they need to not just be totally lame and powergamey. This is what chaplain weapons are now people.

(what happens the remainder 70% to stop spamming)

**Comment by Armhulen** [comment]
Ok??

**Comment by jughu** [comment]
Wasnt the whole purpose chaplain weapons that every weapon had a unique power but also with a downside poisoning yourself sounds like one of then

**Comment by Armhulen** [comment]
Yea

**Comment by FantasticFwoosh** [comment]
You know what i mean @jughu , they aren't cosmetic only, take the carp immunity granted (which co-incides with buffed mega-carp PR Armhulen also made) from the chaplain carp plushy, they age poorly as the game progresses.

This is gimmick weapons gone bad if Armhulen actually went through with merging this trash.

**Comment by jughu** [comment]
That mega carp pr looked more like a nerf to me

**Comment by Armhulen** [comment]
😡 merging 😡 this 😡 trash 😡

**Comment by sawrge** [comment]
stop responding to fwoosh

**Comment by FantasticFwoosh** [comment]
I believe my points to be valid ones, personal opinions of viewing it as 🗑 content aside.

**Comment by DoctoreMad** [comment]
No playtesting is at expense of players, even if it screws the roll that guy that got his yearly traitor roll: It is a lesson that he should just practice Buddhism harder.

**Comment by FantasticFwoosh** [comment]
Just load it on a private lobby, grab a monkeyman and test poisons.

**Review by ShizCalev — DISMISSED** [review]

**Comment by Armhulen** [comment]
I almost forgot, this needs sprites give me a second

**Comment by Armhulen** [comment]
Sprite added asap

**Comment by Armhulen** [comment]
OK super duper ready

**Comment by RogueSteampunker** [comment]
Eh, it seem interesting. If it's now a random chance to actually send the reagents towards them, it would keep Chaps from downing massively harmful chems in the hopes of a ez gg. If it isn't, I'd have to recommend the former.

Honestly though, I don't see too much of an issue. The only major gamebreakers would be slimeperson Chap going around deathcheming everyone, or Chap with hulk bypassing Neurotoxin stuns. And both of those are extremely situational, and could be avoided by running off. He can only have the chems in his system for so long, and the end result is a pretty weak weapon.

And of course, getting the deathchems. Last I checked, most Chemists aren't all that willing to hand out the sort of deathchems that make using this worthwhile, nor do I see a Chap getting enough of em to properly down more than one to two people.

So, in a vacuum, it looks powerful. In practice? It seems like a very niche weapon, that would probably best be utilized in Cult modes, where downing a large deal of Holy Water and beaning your neighborhood cultist with it could easily cause the battle to end nonlethally.

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is currently below zero (-4). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.

**Pull request merged**